wolfcrypt/src/sha256.c

```diff
@@ -1861,13 +1898,20 @@ int wc_Sha256GetHash(wc_Sha256* sha256, byte* hash)
     ret = wc_Sha256Copy(sha256, &tmpSha256);
     if (ret == 0) {
         ret = wc_Sha256Final(&tmpSha256, hash);
-#if  defined(WOLFSSL_ESP32WROOM32_CRYPT) && \
-    !defined(NO_WOLFSSL_ESP32WROOM32_CRYPT_HASH)
+
+#if  defined(WOLFSSL_USE_ESP32WROOM32_CRYPT_HASH_HW)
+    {
         sha256->ctx.mode = ESP32_SHA_SW;
+    }
 #endif
 
         wc_Sha256Free(&tmpSha256);
     }
+
+#if defined(WOLFSSL_USE_ESP32WROOM32_CRYPT_HASH_HW)
+    /* TODO: Make sure the ESP32 crypto mutex is released (in case final is not
```

Comment:
Should we fix this TODO? Should be able to detect and release in the `wc_Sha256Free`

Reply:
I added a check of `sha256->ctx.lockDepth` in `wc_Sha256Free` to ensure the mutex is actually released.